bytes_ostream.hh

```diff
@@ -268,6 +268,43 @@ public:
         write(bytes_view(reinterpret_cast<const signed char*>(ptr), size));
     }
 
+    // Writes the given integer value
+    template <std::integral int_type>
+    void write(const int_type& val) {
+        auto v = net::ntoh(val);
```

Comment:
Actually need hton() since the input is in host byte order.

But I recommend write_be() instead into a byte array so you don't have to deal with the casting.

Reply:
And also, to combine the biasing of signed types with the byteorder conversion, those aren't separate steps.